packages/x-internal-gestures/src/core/PointerManager.ts

```diff
@@ -124,10 +124,10 @@ export class PointerManager {
     new Set();
 
   public constructor(options: PointerManagerOptions) {
-    this.root = (options.root ?? document.documentElement) as HTMLElement;
+    this.root = (options.root ?? document.getRootNode({ composed: true })) as HTMLElement;
```

Comment:
We get the root node instead of `documentElement`, though it shouldn't really change in most cases.